### DISCUSSION TITLE
allow to use environment variable for nextjs rewrites

### DISCUSSION THREAD

**Discussion by jantimon** [vercel/next.js#27041] [Ideas]
### Describe the feature you'd like to request

We would like to use a different fallback server per environment e.g.:

```ts
module.exports = {
  async rewrites() {
    return {
      fallback: [
        // These rewrites are checked after both pages/public files
        // and dynamic routes are checked
        {
          source: '/:path*',
          destination: process.env.FALLBACK_SERVER,
        },
      ],
    }
  }
}
```

Right now this is not possible as `process.env.FALLBACK_SERVER` is processed during build time.  
So we are not able to deploy the same image for different environments 

### Describe the solution you'd like

Allow to evaluate environment variables during runtime e.g. by allowing to define a function for destination:

```ts
module.exports = {
  async rewrites() {
    return {
      fallback: [
        // These rewrites are checked after both pages/public files
        // and dynamic routes are checked
        {
          source: '/:path*',
          destination: () => process.env.FALLBACK_SERVER,
        },
      ],
    }
  }
}
```

### Describe alternatives you've considered

the alternative would be to have different builds per environment.  
however we would prefer to test the exact same build which will later be deployed to production.

**Comment by makeable** [comment]
Just got bitten by this one :(

**Comment by STS-SW** [comment]
How did you solve this?
